### ISSUE TITLE
Auto learning rate finder

### ISSUE THREAD

**Issue by rusmux** [ultralytics/ultralytics#925]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Implement auto learning rate finding when starting training, [like](https://pytorch-lightning.readthedocs.io/en/1.4.5/advanced/lr_finder.html) in PyTorch Lightning. It would help "reduce the amount of guesswork concerning choosing a good initial learning rate".

### Use case

Would be very useful in training models.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by JustasBart** [comment]
@rusmux That sounds awesome! I'd love to see it / try it in the near future!

I'm all in with this :rocket:
